web/src/beta/features/Editor/Tabs/Storytelling/StoryPageIndicator/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { FC } from "react";
+
+import { styled } from "@reearth/services/theme";
+
+type Props = {
+  currentPage: number;
+  currentPageProgress: number;
+  maxPage: number;
+  onChangePage: (page: number) => void;
+};
+
+const StoryPageIndicator: FC<Props> = ({
+  currentPage,
+  currentPageProgress,
+  maxPage,
+  onChangePage,
+}) => {
+  return (
+    <Wrapper>
+      {Array.from({ length: maxPage }).map((_, i) => {
+        const page = i + 1;
+        const isActive = currentPage >= page;
+        const isCurrentPage = page === currentPage;
+        const progress = isCurrentPage ? currentPageProgress : isActive ? 100 : 0;
```

Comment:
I was worried about whether create a function or not and decide to not create it since it's simple enough I guess.
Let me know if you have some ideas.

Reply:
Hmmm, no I think you are right. Since we are just displaying the prop's value a function doesn't really make sense!
Though, please wrap page, isActive, isCurrentPage and progress in useMemo. 
On scroll it will rerender A LOT, right? So we won't want more static stuff like page, isActive, etc to be re-rendered just because currentPageProgess updates

Reply:
I understand your concern and I fixed it. But the code looks a bit more complex than expected. ( Why I wrote like [this](https://github.com/reearth/reearth/pull/543/commits/ac3bc1ed67df7733e90be986044c7f763bd6c318) is memorize `progress` cannot avoid re-render since it depend on `currentPageProgress` . )

So it might be better to think about when we have actual performance issue.

Reply:
Ah that is true about progress, but the other variables are more static and would be nice to be memoized so that when `progress` re-renders the component they don't have to be as well.